composer/algorithms/gyro_dropout/gyro_dropout.py

```diff
@@ -0,0 +1,143 @@
+
+    return GyroDropoutLayer(p, sigma, tau, num_iterations, epoch)
+
+
+def apply_gyro_dropout(model: torch.nn.Module, 
+                       iters_per_epoch: int,
+                       max_epoch: int,
+                       p: float,
+                       sigma: int, 
+                       tau: int) -> None:
+    """Replaces all instances of `torch.nn.Dropout` with a `GyroDropout`.
+
+    By masking Dropout layer, this usually improves accuracy.
+    """
+
+    # prepare the replacement policy and perform replacement
+    from functools import partial
+    policy: Dict[Type[torch.nn.Module], module_surgery.ReplacementFunction] = {torch.nn.Dropout: partial(from_Dropout, iters_per_epoch, max_epoch, p, sigma, tau)}
+    replaced_instances = module_surgery.replace_module_classes(module=model, policies=policy)
+    if len(replaced_instances) == 0:
+        warnings.warn(
+            NoEffectWarning(
+                'No instances of `torch.nn.Dropout` were found, and therefore, there were no modules to replace.'))
+    log.info(f'Successfully replaced {len(replaced_instances)} of dropout with a Gyro dropout.')
+
+
+class GyroDropout(Algorithm):
+    """Replaces all instances of `torch.nn.Dropout` with a `GyroDropout`.
+
+    By masking Dropout layer, this usually improves accuracy.
+
+    Args:
+        p (float, optional): Float number of ratio to dropout.
+            Default: ``0.5``.
+        sigma (int, optional): the number of total pre-selected subnetwork
+            Default: ``256``.
+        tau (int, optional): the number of concurrently scheduled subnetworks in an iteration
+            Default: ``16``.
+            
+    Example:
+           
+        .. testcode::
+
+           from composer.algorithms import GyroDropout
+
+           algorithm = GyroDropout(p=0.5, sigma=256, tau=16)
+           trainer = Trainer(
+               model=model,
+               train_dataloader=train_dataloader,
+               max_duration="100ep",
+               algorithms=[algorithm],
+               optimizers=[optimizer]
+           )
+    """
+
+    def __init__(self, p: float=0.5, sigma: int=256, tau: int=16):
+        self.p = p 
+        self.sigma = sigma
+        self.tau = tau
+        
+    def __repr__(self) -> str:
+        return f'{self.__class__.__name__}()'
+
+    @staticmethod
+    def required_on_load() -> bool:
+        return True
+
+    def match(self, event: Event, state: State) -> bool: 
+        del state
+        return event == Event.FIT_START
+
+    def apply(self, event: Event, state: State, logger: Logger) -> Optional[int]:
+        del event, logger          
+        
+        apply_gyro_dropout(model=state.model,
+                           iters_per_epoch=int(state.dataloader_len),
+                           max_epoch=int(state.max_duration),
+                           p=self.p,
+                           sigma=self.sigma,
+                           tau=self.tau,
+        )
```

Comment:
We deleted parameters -(iters_epoch, max_epoch) by using state. It works in composer's trainer, but user should write iters_per_epoch, max_epoch as parameters in functional api because functional api doesn't support state.